clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -830,7 +843,7 @@ private void handleResponses(long now, List<ClientResponse> responses, Map<Strin
 
                 // Stop tracking this call.
                 correlationIdToCall.remove(correlationId);
-                getOrCreateListValue(callsInFlight, response.requestHeader().clientId()).remove(call);
+                getOrCreateListValue(callsInFlight, response.destination()).remove(call);
```

Comment:
Since we are handling responses here, it seems like there's no point in creating the list value if it doesn't exist, right? Should we just be doing a `get` on the map? Can it ever return a `null` list?